front/app/containers/EmailChange/UpdateEmailForm.tsx

```diff
@@ -57,7 +57,7 @@ const UpdateEmailForm = ({
   const onFormSubmit = async (formValues: FormValues) => {
     try {
       // If confirmation required, launch modal
-      if (appConfiguration?.data?.attributes.settings.user_confirmation) {
+      if (userConfirmationEnabled) {
         resendEmailConfirmationCode(formValues.email)
```

Comment:
Also if that's the case and appConfig.data.attributes.settings.user_confirmation isn't being used anymore, we should change it here as well:
![image](https://github.com/CitizenLabDotCo/citizenlab/assets/33987955/70130c4d-d4a8-44f2-a16a-6da034143d32)
(`/citizenlab/front/app/containers/Admin/settings/registration/index.tsx`)

Reply:
> Previously we also had a toggle in the back office for enabling/disabling email confirmation on the platform right? (so both the toggle and the feature flag had to be enabled) Was that removed then?

I see it locally 
![image](https://github.com/CitizenLabDotCo/citizenlab/assets/91883434/fe42e0e1-22da-4ca7-a2a4-bc97371c6bec)
Not sure why I don't see it here https://demo.stg.citizenlab.co/en/admin/settings/registration

> Also if that's the case and appConfig.data.attributes.settings.user_confirmation isn't being used anymore, we should change it here as well: (/citizenlab/front/app/containers/Admin/settings/registration/index.tsx)

I think `user_confirmation` is still used. But in this example, we only need to check the `enabled` field. 

And in the case in this PR, we need to check both `enabled` and `allowed`, that's why I used `useFeatureFlag`. It would be the same as checking `!!latestAppConfigSettings?.user_confirmation?.enabled && !!latestAppConfigSettings?.user_confirmation?.allowed`.

Reply:
Ahh ok now I understand :) Thanks for explaining!